keystore/keystore_test.go

```diff
@@ -0,0 +1,82 @@
+package keystore
+
+import (
+	"crypto/rand"
+	"testing"
+
+	ccrypto "github.com/libp2p/go-libp2p-core/crypto"
+
+	"github.com/ipfs/go-datastore"
+)
+
+func TestSignAndVerify(t *testing.T) {
+	const Message = "Test Message"
+	ds := datastore.NewNullDatastore()
+	ks, err := NewKeystore(ds)
+	if err != nil {
+		t.Error(err)
+	}
+	priv, pub, _ := ccrypto.GenerateRSAKeyPair(2048, rand.Reader)
+	signed, err := ks.Sign(priv, []byte(Message))
+	if err != nil {
+		t.Error(err)
+	}
+
+	if err := ks.Verify(signed, pub, []byte(Message)); err != nil {
+		t.Error()
+	}
+}
+
+func TestHasKey(t *testing.T) {
+	ds := datastore.NewNullDatastore()
+	ks, err := NewKeystore(ds)
+	if err != nil {
+		t.Error(err)
+	}
+	_, err = ks.CreateKey("test_key_1")
+	if err != nil {
+		t.Error(err)
+	}
+
+	exists, err := ks.HasKey("test_key_1")
+	if err != nil {
+		t.Error(err)
+	}
+	if !exists {
+		t.Errorf("failed to find key in cache")
+	}
```

Comment:
This only works because the Keystore has a *lru.Cache accessed before reaching the Datastore